### PR TITLE
Int b 19814 add origin/dest sit auth end date to db

### PR DESCRIPTION
**Pull request by MichaelEngelbertCACI**
## Summary

This Pull request adds the columns origin_sit_auth_end_date and dest_sit_auth_end_date to the mto_shipments table. These dates will be calculated as the day the shipment enters origin/dest SIT + the SIT day allowance (DOFSIT/DDFIST + sit_days_allowance). Storing these values as columns in the database will reduce the need of computing these values on the frontend/backend and instead allow them to be statically held in the db and allow Prime to request an endpoint for the values when needed. 

## Verification Steps for the Author

These are to be checked by the author.

- [v] Tested in the local environment
- [v] Have the Agility acceptance criteria been met for this change?
- [v] Columns are added to the mto_shipments table in the dev_db

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. run git fetch --all in terminal of mymove repo to fetch all updated branches.
2. git checkout integrationTesting
3. git pull
4. git checkout INT-B-19814-add-origin/dest-sit-auth-end-date-to-db to checkout the code from this PR 
5. make db_dev_run
6. make db_dev_fresh db_dev_migrate to run the newly added migration script
7. Once the scripts have ran open up dbeaver and connect to the database and navigate to the mto_shipments table and verify origin_sit_auth_end_date and dest_sit_auth_end_date have been added as columns to the table as a date datatype. 

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-05-13 at 11 33 22 AM

### CODE REVIEW

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Couple of things we like to have when adding columns/tables via migration files.

- Throwing in `IF NOT EXISTS` before the column name. This helps us avoid some unnecessary errors when running the migration.
ex. `ADD COLUMN  IF NOT EXISTS origin_sit_auth_end_date`

- Also make sure you're adding comments to the added columns - more info means more understanding.
ex. 
```
COMMENT ON COLUMN mto_shipments.origin_sit_auth_end_date IS 'This is a comment for this column.';
```

You can find some good examples in the existing migration files.

Some things specific to this implementation:
- Could you change the column names to use `auth` instead of `aut`? It is short for `authorized` so we want that to reflect accurately in the shorthand (also it says in the title of the BL item). Also keep in mind you'll need to rename the title of the migration file as well.

- In regards to adding comments for the columns. Are you familiar with SIT and what it is? What the authorized end date is? What the difference between origin and destination SIT is? Doing this small PRs like this is a great time to learn what they are so your comments can accurately reflect what the column is for.

- The PR description should not remain a boilerplate. Please fill in all relevant information, how to test as reviewers, plenty of PRs for you to compare to in order to keep it similar to how we've been doing it.


As always... Holler if you have questions. I invited you to a Teams meeting next Monday to go over all of these things, but please use your time between now and then to work on changing these things and improving your understanding on WHAT these changes mean and how it impacts the application. Kudos to getting your first PR up!

**Comment by danieljordan-caci** [comment]
In addition to what I just put, make sure you're:
- assigning yourself to this PR in the `Assignees` section
- when requesting to merge code into `integrationTesting`, we want to add the label `INTEGRATION` to the PR
- always add the label of the team for your PRs, you can search for `Mountain Movers` and it's pink

**Review by traskowskycaci — APPROVED** [review]
I noticed aut rather than auth in the migration title, do you mind fixing that?

I do see the columns locally.

Make sure to get the summary of your PR update and also merge in the latest integration before pushing up.

**Review by cameroncaci — COMMENTED** [review]
Migration worked like a charm. @traskowskycaci @danieljordan-caci thoughts on if we remove the abbreviations and replace it with the long form of `origin_sit_authorized_end_date` and `destination_sit_authorized_end_date`? This would update "auth" and "dest" with their full wordage

**Comment by danieljordan-caci** [comment]
> Migration worked like a charm. @traskowskycaci @danieljordan-caci thoughts on if we remove the abbreviations and replace it with the long form of `origin_sit_authorized_end_date` and `destination_sit_authorized_end_date`? This would update "auth" and "dest" with their full wordage

I don't feel it is necessary, personally - not against it though. I know that's not helpful, but that's how I feel haha

**Comment by traskowskycaci** [comment]
> Migration worked like a charm. @traskowskycaci @danieljordan-caci thoughts on if we remove the abbreviations and replace it with the long form of `origin_sit_authorized_end_date` and `destination_sit_authorized_end_date`? This would update "auth" and "dest" with their full wordage

Either way works for me!

File: migrations/app/schema/20240503181242_add_origin_dest_sit_aut_end_date_to_mto_shipments_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE mto_shipments
+    ADD COLUMN IF NOT EXISTS origin_sit_auth_end_date DATE,
+    ADD COLUMN IF NOT EXISTS dest_sit_auth_end_date DATE;
+
+COMMENT ON COLUMN mto_shipments.origin_sit_auth_end_date IS 'receive the initial SIT authorized end date from the origin and subsequent updates';
+COMMENT ON COLUMN mto_shipments.dest_sit_auth_end_date IS 'receive the initial SIT authorized end date from the destination and subsequent updates';
```
Comment (danieljordan-caci) [inline migrations/app/schema/20240503181242_add_origin_dest_sit_aut_end_date_to_mto_shipments_table.up.sql:6]: Please change the title of this migration from to use `auth` and not `aut`

**Review by danieljordan-caci — APPROVED** [review]
I see the description is updated! Nice. Just one more little thing and this should be good to go.

**Pull request merged**